lib/mix/test/mix/cli_test.exs

```diff
@@ -19,6 +19,40 @@ defmodule Mix.CLITest do
     end)
   end
 
+  test "Mix.raise/2 can set exit code", %{tmp_dir: tmp_dir} do
+    File.cd!(tmp_dir, fn ->
+      File.mkdir_p!("lib")
+
+      File.write!("mix.exs", """
+      defmodule MyProject do
+        use Mix.Project
+
+        def project do
+          [app: :my_project, version: "0.0.1", aliases: aliases()]
+        end
+
+        defp aliases do
+          [
+            regular: &error(&1, []),
+            custom: &error(&1, code: 99),
+            too_high: &error(&1, code: 1000),
+            no_error: &error(&1, code: 0),
+            negative: &error(&1, code: -1)
+          ]
+        end
+
+        defp error(_args, opts), do: Mix.raise("oops", opts)
+      end
+      """)
+
+      assert {_, 1} = mix_code(~w[regular])
+      assert {_, 0} = mix_code(~w[no_error])
+      assert {_, 99} = mix_code(~w[custom])
+      assert {_, 1} = mix_code(~w[too_high])
+      assert {_, 1} = mix_code(~w[negative])
+    end)
```

Comment:
Those tests are very expensive so I think we are being too extensive in our coverage. I think a simple test that raises a custom code, such as `13` or whatever, should be enough.